opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/TimeData.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.opentaskspal.utils.TimeZoneEqualator;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * {@link RowData} for adding start, start+due, or start+duration times.
+ *
+ * @author Gabor Keszthelyi
+ */
+// TODO Unit test
+public final class TimeData implements RowData<TaskContract.Tasks>
+{
+    @NonNull
+    private final DateTime mStart;
+    @Nullable
+    private final DateTime mDue;
+    @Nullable
+    private final String mDuration;
+
+
+    private TimeData(@NonNull DateTime start, @Nullable DateTime due, @Nullable String duration)
+    {
+        mStart = start;
+        mDue = due;
+        mDuration = duration;
+    }
+
+
+    public TimeData(@NonNull DateTime start, @NonNull DateTime due)
+    {
+        this(start, due, null);
+    }
+
+
+    public TimeData(@NonNull DateTime start, @NonNull String duration)
+    {
+        this(start, null, duration);
+    }
+
+
+    public TimeData(@NonNull DateTime start)
+    {
+        this(start, null, null);
+    }
+
+
+    @NonNull
+    @Override
+    public ContentProviderOperation.Builder updatedBuilder(@NonNull TransactionContext transactionContext, @NonNull ContentProviderOperation.Builder builder)
+    {
+        if (mDue != null && mStart.isAllDay() != mDue.isAllDay())
+        {
+            throw new IllegalArgumentException("'start' and 'due' must have the same all-day flag");
+        }
+
+        DateTime start = mStart;
+        if (mDue != null && !TimeZoneEqualator.INSTANCE.areEqual(mStart, mDue))
```

Comment:
There is no need for this Equalator. DateTime is smart enough to return the same instance if the timezone is considered to be the same.

Reply:
You mean that there is no need to compare the `getId()` `String` value but use reference equality on the `getTimeZone()` `TimeZone`s? Like `if (timeZone1 == timeZone2)`?
Don't we depend on some undocumented internals of `DateTime` this way?
I've updated the code, removed the `Equalator` and added a comment about this behavior, but I think it would be clearer to check the `getId()` values. `TimeZone` doesn't provide `equals()`, so we are really depending on reference equality here which doesn't seem to be safe.
I can do that in private method if you are not consent with that `Equalator`.
(Although it could be easier to test that code in isolation and ideally just mock it out in (upcoming) `TimeZoneTest` if it could be injected there, but it couldn't.. so it may not be important.)

Reply:
No that's not what I meant. I've added a new comment to the new code. The comment above was a bit misleading.

Reply:
(addressed)